src/redux/handlers/onStart.js

```diff
@@ -1,27 +1,15 @@
 import Immutable from 'seamless-immutable';
-import { deckComponents } from '../../constants/settings';
-import { createPlayer } from '../../models/player';
-import { createRandomPosition } from '../../models/position';
-import { createDeck } from '../../models/card';
-import { playersToObject } from './common';
+import { head } from 'ramda';
+import { createPlayerEntities } from '../../models/player';
 
 export const onStart = (state, action) => {
-  const boardMaxIdx = state.boardSize - 1;
-  const users = action.payload.users;
-  const players = users.map((user, idx) =>
-    createPlayer(user, {
-      cards: createDeck(deckComponents),
-      position: createRandomPosition(0, boardMaxIdx, idx * 5, idx * 5 + 2),
-    })
-  );
-
-  const entities = playersToObject(players);
+  const entities = createPlayerEntities(state.boardSize, action.payload.users);
 
   return Immutable.merge(state, {
     isEnd: false,
     entities,
     roundCounter: 0,
     roundMoves: 0,
-    turn: players[0].id,
+    turn: Number(head(Object.keys(entities))),
```

Comment:
the entities object will contain all objects from the board, not only players. there was distinction between players and entities and it should stay imho

Reply:
and maybe is worth to get random idx instead of 0.

Reply:
I understand and agree - but here 'entities' refers only to player entities whose keys are equal to player ids, see line 6:
`const entities = createPlayerEntities(state.boardSize, action.payload.users);`

Reply:
you referring to piece of code you just wrote. well, yes, the only entity is used is player for now but that's temporary and this kind of assumption in your code telling that entites = players